ui/src/app/views/settings/component.spec.ts

```diff
@@ -0,0 +1,77 @@
+import {ComponentFixture, fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {By} from '@angular/platform-browser';
+import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
+import {RouterTestingModule} from '@angular/router/testing';
+import {ClarityModule} from '@clr/angular';
+
+import {SettingsComponent} from 'app/views/settings/component';
+import {ClusterServiceStub} from 'testing/stubs/cluster-service-stub';
+
+import {ClusterService} from 'generated';
+
+
+describe('SettingsComponent', () => {
+  let fixture: ComponentFixture<SettingsComponent>;
+  let clusterService: ClusterServiceStub;
+
+  beforeEach(fakeAsync(() => {
+    clusterService = new ClusterServiceStub();
+    TestBed.configureTestingModule({
+      declarations: [
+        SettingsComponent
+      ],
+      imports: [
+        BrowserAnimationsModule,
+        RouterTestingModule,
+        ClarityModule.forRoot()
+      ],
+      providers: [
+        { provide: ClusterService, useValue: clusterService },
+      ]}).compileComponents();
+  }));
+
+  beforeEach(fakeAsync(() => {
+    fixture = TestBed.createComponent(SettingsComponent);
+    // Trigger ngOnInit, tick() the promise, then update again to undisable the
+    // "reset" button.
+    fixture.detectChanges();
+    tick();
+    fixture.detectChanges();
+  }));
+
+  it('should open the cluster reset modal', fakeAsync(() => {
+    const de = fixture.debugElement;
+    expect(de.queryAll(By.css('.modal-body')).length).toEqual(0);
+    de.query(By.css('.notebook-settings button')).nativeElement.click();
```

Comment:
Similarly can use the query helper functions. (Sort of abstracting away some of the specific syntax of this stuff.)

Reply:
I don't agree with those two helpers. I'd prefer to write normal Angular testing code rather than using custom wrappers, unless there is significant pain with the standard approach; I'd say this doesn't pass that threshold . The standard utilities are more general, have more complete documentation, and can be learned and applied to any Angular project. The helper saves a few characters, but adds the burden of a layer of indirection.

`simulateClick()` is somewhat useful because it also updates (though this is perhaps slightly surprising), I will use that here.

Reply:
That's fair. I think I agree with you, although the ramp up time on learning some of the specific angular testing code was sort of difficult, which is why I abstracted them. (I had trouble finding the actual documentation on how they actually worked, and generally found it confusing.)

I think I'd be fine with moving back to the angular methods, but that is with the understanding that there is likely to be some similar problems for people who haven't worked in this framework before.

Reply:
Agreed it's not easy to learn. I think this can be largely improved by having good example tests in the codebase. Ultimately, most people will copy-paste an existing test and tweak from there to bootstrap things anyways.